ledger/src/blockstore.rs

```diff
@@ -1395,11 +1396,47 @@ impl Blockstore {
                 continue;
             }
 
-            self.parent_meta_cf.put_in_batch(
-                &mut shred_insertion_tracker.write_batch,
-                (slot, location),
-                working_parent_meta.as_ref(),
-            )?;
+            let key = (slot, location);
+
+            let WorkingEntry::Dirty(new_parent_meta) = working_parent_meta else {
```

Comment:
This logic is better placed when we actually update the parent meta rather than in this function where we are committing batches. You can grab the current `ParentMeta` before you call the `maybe` functions.

See the `MerkleRootMeta` logic for an example:
```rust
if let HashMapEntry::Vacant(entry) = merkle_root_metas.entry((location, erasure_set)) {
            if let Some(meta) = self
                .merkle_root_meta_from_location(erasure_set, location)
                .unwrap()
            {
                entry.insert(WorkingEntry::Clean(meta));
            }
        }
```

Reply:
Actually I think it's even better if we don't enforce this here - if `ParentMeta` is already populated from `UpdateParent` then don't bother calling `maybe_insert_update_parent`. Instead the `block_component_processor` will mark dead if we see multiple update parents - saves us a bunch of blockstore lookups.